modules/javafx.graphics/src/test/java/test/javafx/scene/Node_lookup_Test.java

```diff
@@ -96,4 +109,41 @@ public class Node_lookup_Test {
         assertTrue(nodes.contains(d));
         assertTrue(nodes.contains(d2));
     }
+
+    @Test
+    public void lookupPseudoTest(){
+        Set<Node> nodes = root.lookupAll(".h:testPseudo2");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        Node found = root.lookup(".h:testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(":testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(".h:testPseudo1:testPseudo2");
+        assertSame(hg, found);
+
+        nodes = root.lookupAll(".g:testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(":testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
```

Comment:
I'd add one more test case to verify lack of regression when no pseudo class is specified.  May be something like this (feel free to expand, add negative cases etc.):

```
    /**
     * Verifies that the lookup ignores pseudo classes when selector contains no explicit pseudo class.
     */
    @Test
    public void lookupPseudoTest2() {
        Group root = new Group();
        
        Group ab = new Group();
        ab.getStyleClass().addAll("a", "b");
        
        Group a = new Group();
        a.getStyleClass().addAll("a");
        
        Group a1 = new Group();
        a1.getStyleClass().addAll("a");
        a1.pseudoClassStateChanged(PseudoClass.getPseudoClass("P1"), true);
        
        ParentShim.getChildren(root).addAll(a, a1, ab);
        
        Set<Node> rv;
        
        rv = root.lookupAll(".a");
        assertTrue(rv.contains(a));
        assertTrue(rv.contains(a1));
        
        rv = root.lookupAll(".a:P1");
        assertTrue(rv.contains(a1));
    }
```

Reply:
Added another test case to cover the scenario you mentioned. I used the same nodes (have pseudo classes set)  which are used in previous test case. Tried to cover all possible scenarios to check for regression.